pegen/c_generator.py

```diff
@@ -145,6 +146,31 @@ class CParserGenerator(ParserGenerator, GrammarVisitor):
     def __init__(self, rules: Dict[str, grammar.Rule], file: Optional[IO[Text]]):
         super().__init__(rules, file)
         self.callmakervisitor = CCallMakerVisitor(self)
+        self._varname_counter = itertools.count()
+
+    def unique_varname(self, name="tmpvar"):
+        return name + "_" + str(next(self._varname_counter))
+
+    def call_with_errorcheck(self, call_text, returnval="", goto_target=None):
+        error_var = self.unique_varname()
+        self.print(f"int {error_var} = {call_text}")
```

Comment:
I find it jarring that the `;` has to be passed in -- maybe generate it here and remove it from the call site.